commons/ava_services/gecko_service_availability_checker_core.go

```diff
@@ -18,20 +24,15 @@ func (g GeckoServiceAvailabilityCheckerCore) IsServiceUp(toCheck services.Servic
 	client := gecko_client.NewGeckoClient(jsonRpcSocket.GetIpAddr(), jsonRpcSocket.GetPort())
 	healthInfo, err := client.HealthApi().GetLiveness()
 	if err != nil {
-		logrus.Trace(stacktrace.Propagate(err, "Error occurred in getting liveness info"))
+		// TODO increase log level when InitialDelay is set to a conservative value
+		logrus.Info(stacktrace.Propagate(err, "Error occurred in getting liveness info"))
 		return false
 	}
+	logrus.Infof("Service reported healthy: %v", healthInfo.Healthy)
 
-	// HACK HACK HACK we need to wait for bootstrapping to finish, and there is not API for this yet (in development)
-	// TODO TODO TODO once bootstrapping checker is available, use that instead of just waiting
-	if healthInfo.Healthy {
```

Comment:
I've never run into this being a problem before, but you're right once it's done bootstrapping it will add the API handlers, so this could cause a problem. I'll add the sleep statement back in to handle this case.